### ISSUE TITLE
[UI] Capture TurboBadger hashtable fix from repo

### ISSUE THREAD

**Issue by JoshEngebretson** [AtomicGameEngine/AtomicGameEngine#1205]
There may be other updates in the TB repo that can be rolled in, this hashtable one would be good:

https://github.com/fruxo/turbobadger/commit/898cc72c1c3e771b62aed887a7ee53a6ffb421dd

**Comment by JimMarlowe** [comment]
I'll apply all PRs up to the current point in time. It looks like most of the PRs are from one guy, and there is not a lot of new functionality.

**Comment by JimMarlowe** [comment]
Doing a clean reboot of TB is a difficult task, because of the changes required for scripting support. It just means new features will need to be hand merged. The TB rate of change has been small, so this hasn't been an issue.

**Comment by JimMarlowe** [comment]
Upgraded turbobadger some, it includes commits/prs:
ee05368 API for multitouch
898cc72 Added some unit tests & fixed bug in TBHashTable.
TBWindow should have a virtual destructor 102
Fix TBImage::IsEmpty 93

I had to do some additional fixes for ee05368, I saw during compile that InvokePointerDown() and InvokePointerUp() were expecting to return bool, but they were still returning void, so I fixed them to the spirit in which they were intended.  This means that both touch systems will be installed, and It will be possible to transition from one to the other.

Tested on Android as best as I could, I tried the Roboman example, but it is crashing out on all my devices with 
```
E/Urho3D  ( 2721): JSErrorEvent:  : Line 90
E/Urho3D  ( 2721):  Name: TypeError
E/Urho3D  ( 2721):  Message: undefined not callable
E/Urho3D  ( 2721):  Stack:TypeError: undefined not callable
E/Urho3D  ( 2721): 	duk_js_call.c:776
E/Urho3D  ( 2721): 	anon DPad:90
E/Urho3D  ( 2721): 	anon DPad:58
E/Urho3D  ( 2721): 	anon Scripts/main:9 preventsyield
E/Urho3D  ( 2721): 	require  native strict preventsyield
D/dalvikvm( 2721): threadid=15: thread exiting, not yet detached (count=0)
```
I tried it also with an earlier Atomic version, and I get the same crash, so it's not these changes that makes it happen. I tried some of my own programs that use TB+touch for UI on android, an they worked correctly.

**Comment by JoshEngebretson** [comment]
Ok, that example dpad JS code is crusty and uses some multi-touch code which was introduced into TB some time ago. It could conflict in some way with new implementation from the TB repo.  The script is attempting to call an undefined function.  I'll file an issue to look into it.

**Comment by JoshEngebretson** [comment]
Closed via #1246
